api/src/main/java/org/openmrs/module/Module.java

```diff
@@ -530,11 +530,15 @@ public void setExtensionNames(Map<String, String> map) {
 
 	private boolean isNoNeedToExpand() {
 		if (extensionNames == null || extensionNames.isEmpty()) {
-			return true;
+			for(Extension ext : extensions) {
+				if (!extensionNames.get(ext.getPointId()).equals(ext.getClass().getName())) {
```

Comment:
Why try to get elements of extensionNames if the if already confirmed that is null or empty? In best case nothing happen and in worst case you get nullPointerException .